modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```
@@ -87,30 +92,34 @@ public int getSize() {
    * @return
    */
   private <R> R performWithConnection(Function<Object, R> function) {
-    ConnectionSupplier connectionSupplier = getConnectionSupplier();
+    connectionSupplier = getConnectionSupplier();
     Object connection = getConnection(connectionSupplier);
     try {
       R result = function.apply(connection);
       connectionSupplier.close();
       return result;
     } catch (Exception e) {
       if (isFirstPage) {
-        closeDelegate(connection);
+        silentlyCloseDelegate(connection);
       }
       extractConnectionException(e).ifPresent(ex -> connectionSupplier.invalidateConnection());
```

still not fixed. If this fails, the original exception get swallowed